lib/dogma/rules/trailing_whitespace.ex

```diff
@@ -13,6 +13,23 @@ defmodule Dogma.Rules.TrailingWhitespace do
     Enum.reduce( script.processed_lines, [], &check_line(&1, &2) )
   end
 
+  def correction(source, errors) do
+    lines = Enum.map(errors, fn err -> err.line end)
+    [newline] = Regex.run(~r/\r?\n\z/, source)
+
+    source
+    |> String.split(newline)
+    |> Enum.with_index
+    |> Enum.map(fn {line, i} ->
+      if Enum.member?(lines, i + 1) do
+        String.replace(line, @regex, "")
+      else
+        line
+      end
+    end)
+    |> Enum.join(newline)
```

Comment:
By only fixing lines where the error was detected we avoid stripping strings -- unless there is an edge case where that isn't true.

I think you're right that this can be simplified with a better regex, though.